https://issues.redhat.com/browse/ACM-15868 Add External platform (#7362)
open-cluster-management/rhacm-docs: clusters/cluster_lifecycle/create_cluster_on_prem.adoc

[#creating-a-cluster-on-premises]
= Creating a cluster in an on-premises environment

> You can use the console to create on-premises {ocp} clusters. The clusters can be {sno} clusters, multi-node clusters, and compact three-node clusters on VMware vSphere, Red Hat OpenStack, Nutanix, External, or in a bare metal environment.

Reviewer: What do you think about making it a bit more user focused and action oriented? Something like: 

```suggestion
You can use the console to create on-premises {ocp} clusters. You can create {sno} clusters, multi-node clusters, and compact three-node clusters. You can create these clusters on VMware vSphere, Red Hat OpenStack, Nutanix, External, or in a bare metal environment.
```

> There is no platform integration with the platform where you install the cluster, as the platform value is set to `platform=none`. A {sno} cluster has only a single node, which hosts the control plane services and the user workloads. This configuration can be helpful when you want to minimize the resource footprint of the cluster. 

Reviewer: I think it might read a bit easier if we flip the order of this first sentence, what do you think? Something like: 

```suggestion
Since the platform value is set to `platform=none`, there is no platform integration with the platform where you install the cluster. A {sno} cluster has only a single node, which hosts the control plane services and the user workloads. This configuration can be helpful when you want to minimize the resource footprint of the cluster. 
```

Reviewer: What do you think about make the last sentence more authoritative by making it a statement of what users do rather than simply saying it's helpful? 

```suggestion
There is no platform integration with the platform where you install the cluster, as the platform value is set to `platform=none`. A {sno} cluster has only a single node, which hosts the control plane services and the user workloads. Use this configuration when you want to minimize the resource footprint of the cluster. 
```

<1> Specify the platform type of the environment where the cluster is created. Valid values are: `BareMetal`, `None`, `VSphere`, `Nutanix`, or `External`.  
<2> Use the same name that you used for your `ClusterDeployment` resource.
<3> Use the `ClusterImageSet` that you generated in <<on-prem-creating-your-cluster-with-the-cli-cluster-image-set,Generate a ClusterImageSet>>.
<4> You can specify your SSH public key, which enables you to access the host after it is installed.

If you set `platformType:` to `External`, you must add the `external` spec. See the following example:

[source,yaml]
----
apiVersion: extensions.hive.openshift.io/v1beta1
kind: AgentClusterInstall
metadata:
  name: test-agent-cluster-install
  namespace: demo-worker4
spec:
  platformType: External
  external:
    platformName: "myplatform" <1>
    cloudControllerManager: External <2>
----

<1> Add your platform name.

> <2> Leave empty to disable the Cloud Controller Manager during installation. If set to `External`, you must add custom manifests to enable the cloud controller manager.

Reviewer 2: ```suggestion
<2> Leave empty to disable the Cloud Controller Manager during installation. If set to `External`, you must add custom manifests to enable the Cloud Controller Manager.
```

nit

Reviewer: Not sure if I'm understanding the second bullet here. Do we mean something like the following?: 

 ```suggestion
<2> Leave `empty` as it is so that is disables the Cloud Controller Manager during installation. If set to `External`, you must add custom manifests to enable the Cloud Controller Manager.
```

Author: For this bullet we can't use `empty` since it's not a valid option. We can just say leave it empty or maybe say unpopulated, though that word might be less accessible.

> <2> Leave empty to disable the Cloud Controller Manager during installation. If set to `External`, you must add custom manifests to enable the Cloud Controller Manager.

Reviewer: ```suggestion
<2> The field is automatically empty, `{ }`, so it disables the Cloud Controller Manager during installation. If you set it to `External`, you must add custom manifests to enable the Cloud Controller Manager.
```

Reviewer: ```suggestion
<2> If you set the field to empty, `{}`, it disables the Cloud Controller Manager during installation. If you set it to `External`, you must add custom manifests to enable the Cloud Controller Manager.
```

> <2> Leave empty to disable the Cloud Controller Manager during installation. If set to `External`, you must add custom manifests to enable the Cloud Controller Manager.

Author: ```suggestion
<2> Set the field to empty (`{ }`) to disable the Cloud Controller Manager during installation. If you set it to `External`, you must add custom manifests to enable the Cloud Controller Manager.
```

<1> Specify the platform type of the environment where the cluster is created. Valid values are: `BareMetal`, `None`, `VSphere`, `Nutanix`, or `External`.  
<2> Use the same name that you used for your `ClusterDeployment` resource.
<3> Use the `ClusterImageSet` that you generated in <<on-prem-creating-your-cluster-with-the-cli-cluster-image-set,Generate a ClusterImageSet>>.
<4> You can specify your SSH public key, which enables you to access the host after it is installed.

If you set `platformType:` to `External`, you must add the `external` spec. See the following example:

[source,yaml]
----
apiVersion: extensions.hive.openshift.io/v1beta1
kind: AgentClusterInstall
metadata:
  name: test-agent-cluster-install
  namespace: demo-worker4
spec:
  platformType: External
  external:
    platformName: "myplatform" <1>
    cloudControllerManager: External <2>
----

<1> Add your platform name.

> <2> Leave empty to disable the Cloud Controller Manager during installation.

Reviewer 2: ```suggestion
<2> Leave empty to disable the Cloud Controller Manager during installation.
```

what do you think about adding that when `cloudControllerManager` is set to `External`, we expect the user to add custom manifests that will enable the cloud controller manager.

Author: I added your suggestion, let me know if it looks good now. Thanks!
